File: module/Utils/ZstdTool.py
import compression.zstd


class ZstdTool:
    """Zstd 压缩/解压工具类。"""

    # 压缩级别（1-22，默认 3 是速度与压缩率的平衡点）
    COMPRESSION_LEVEL = 3

    @classmethod
    def compress(cls, data: bytes) -> bytes:
        """压缩数据"""
        return compression.zstd.compress(data, level=cls.COMPRESSION_LEVEL)

    @classmethod
    def decompress(cls, data: bytes) -> bytes:
        """解压数据"""
        return compression.zstd.decompress(data)

File: frontend/src/main/utils/zstd-tool.ts
import { constants, zstdCompressSync, zstdDecompressSync } from "node:zlib";

// 固定 .lg asset 压缩等级，避免新旧工程物理格式参数漂移。
const COMPRESSION_LEVEL = 3;

export class ZstdTool {
  public static readonly COMPRESSION_LEVEL = COMPRESSION_LEVEL;

  public static isRuntimeAvailable(): boolean {
    return (
      typeof zstdCompressSync === "function" &&
      typeof zstdDecompressSync === "function" &&
      typeof constants.ZSTD_c_compressionLevel === "number"
    );
  }

  public static compress(data: Buffer): Buffer {
    return zstdCompressSync(data, {
      params: {
        [constants.ZSTD_c_compressionLevel]: this.COMPRESSION_LEVEL,
      },
    });
  }

  public static decompress(data: Buffer): Buffer {
    return zstdDecompressSync(data);
  }
}
